assets/js/src/segments/dynamic/dynamic_segments_filters/woocommerce.tsx

```diff
@@ -70,21 +85,67 @@ export const WooCommerceFields: React.FunctionComponent<Props> = ({ onChange, it
         </div>
       </div>
     );
-  }
+  } else if (item.action === WooCommerceActionTypes.PURCHASED_CATEGORY) {
+    optionFields = (
+      <div className="mailpoet-form-field">
+        <div className="mailpoet-form-input mailpoet-form-select">
+          <Select
+            placeholder={MailPoet.I18n.t('selectWooPurchasedCategory')}
+            options={categoryOptions}
+            value={find(['value', item.category_id], categoryOptions)}
+            onChange={(option: SelectOption): void => compose([
+              onChange,
+              assign(item),
+            ])({ category_id: option.value })}
+          />
+        </div>
+      </div>
+    );
+  } else if (item.action === WooCommerceActionTypes.NUMBER_OF_ORDERS) {
+    if (!item.number_of_orders_type) {
+      item.number_of_orders_type = '=';
+    }
 
-  return (
-    <div className="mailpoet-form-field">
-      <div className="mailpoet-form-input mailpoet-form-select">
-        <Select
-          placeholder={MailPoet.I18n.t('selectWooPurchasedCategory')}
-          options={categoryOptions}
-          value={find(['value', item.category_id], categoryOptions)}
-          onChange={(option: SelectOption): void => compose([
-            onChange,
-            assign(item),
-          ])({ category_id: option.value })}
-        />
+    optionFields = (
+      <div>
+        <Grid.CenteredRow className="mailpoet-form-field">
+          <Select
+            options={numberOfOrdersTypeOptions}
+            value={find(['value', item.number_of_orders_type], numberOfOrdersTypeOptions)}
+            onChange={(option: SelectOption): void => compose([
+              onChange,
+              assign(item),
+            ])({ number_of_orders_type: option.value })}
+          />
+          <Input
+            type="number"
+            min={0}
+            value={item.number_of_orders_count || ''}
+            placeholder="count"
+            onChange={(event): void => compose([
```

Comment:
Similar to the comment above, I don't fully understand what is happening inside this `onChange`. I made some modifications from the one used above in the Select component and it seems to work. But I don't know if there is a better way to do this.